UnitySDK/Assets/ML-Agents/Scripts/IBrain.cs

```diff
@@ -0,0 +1,23 @@
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace MLAgents
+{
+    /// <summary>
+    /// Brain receive data from Agents through calls to SubscribeAgentForDecision. The brain then updates the
+    /// actions of the agents at each FixedUpdate.
+    /// The Brain encapsulates the decision making process. Every Agent must be assigned a Brain,
+    /// but you can use the same Brain with more than one Agent. You can also create several
+    /// Brains, attach each of the Brain to one or more than one Agent.
+    /// Brain assets has several important properties that you can set using the Inspector window.
```

Comment:
A lot of the things mentioned here don't correspond to anything in this interface.  For instance 
- inspector properties.
- Action and Observation space sizes.

Reply:
Yes, Thank you. This PR does not contain any change to the Documentation or the comments and you will notice I commented out the unit tests. I just want to see if the idea makes sense and how it feels to use compare to the current version.